shell/platform/linux/fl_engine.cc

```diff
@@ -210,13 +235,17 @@ FlEngine* fl_engine_new(FlDartProject* project, FlRenderer* renderer) {
   g_return_val_if_fail(FL_IS_DART_PROJECT(project), nullptr);
   g_return_val_if_fail(FL_IS_RENDERER(renderer), nullptr);
 
-  FlEngine* self =
-      static_cast<FlEngine*>(g_object_new(fl_engine_get_type(), nullptr));
-  self->project = static_cast<FlDartProject*>(g_object_ref(project));
-  self->renderer = static_cast<FlRenderer*>(g_object_ref(renderer));
+  FlEngine* self = FL_ENGINE(g_object_new(fl_engine_get_type(), nullptr));
+  self->project = FL_DART_PROJECT(g_object_ref(project));
+  self->renderer = FL_RENDERER(g_object_ref(renderer));
   return self;
 }
 
+G_MODULE_EXPORT FlEngine* fl_engine_new_headless(FlDartProject* project) {
+  g_autoptr(FlRendererHeadless) renderer = fl_renderer_headless_new();
+  return fl_engine_new(project, FL_RENDERER(renderer));
```

Comment:
> I copied the existing Linux implementation and made the `FlEngine` be a `FlPluginRegistry` with a new `fl_engine_new_headless` constructor - is this the intended architecture?

The way macOS/iOS work is that the view is a settable property of the engine. Here, the view/renderer distinction makes it less clear how that should work; it seems like we only need this special API because unlike macOS/iOS view objects, which are public, the GTK engine is using renderer which is an internal implementation detail.

I'm okay starting with this though, and we can see what kind of feedback we get on it post-alpha, and then iterating from there as necessary.